processor/src/lib.rs

```diff
@@ -502,6 +502,9 @@ where
             Decorator::Event(id) => {
                 self.host.borrow_mut().on_event(self, *id)?;
             }
+            Decorator::Trace(id) => {
+                self.host.borrow_mut().on_trace(self, *id)?;
+            }
```

Comment:
Should we not check here if tracing is enabled and call `on_trace()` only if it is?